api/v1beta1/verticareplicator_types.go

```diff
@@ -43,6 +43,45 @@ type VerticaReplicatorSpec struct {
 	// be enabled on the source database. Custom username for source and target
 	// databases is not supported yet when TLS configuration is used.
 	TLSConfig string `json:"tlsConfig,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=async
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// Information of the target Vertica database to replicate to
+	Mode string `json:"mode"`
+}
+
+type VerticaReplicatorSourceDatabaseInfo struct {
+	VerticaReplicatorDatabaseInfo `json:",inline"`
+
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
+	// The object name we want to copy from the source side. The available types are: namespace, schema, table.
+	// If this is omitted, the operator will replicate all namespaces in the source database.
+	ObjectName string `json:"objectName"`
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
+	// A string containing a wildcard pattern of the schemas and/or tables to include in the replication.
+	// Namespace names must be front-qualified with a period.
+	IncludePattern string `json:"includePattern"`
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
+	// A string containing a wildcard pattern of the schemas and/or tables to exclude from the set of tables matched
+	// by the include pattern. Namespace names must be front-qualified with a period.
+	ExcludePattern string `json:"excludePattern"`
+}
+
+type VerticaReplicatorTargetDatabaseInfo struct {
+	VerticaReplicatorDatabaseInfo `json:",inline"`
+
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
+	// Namespace in the target database to which objects are replicated. The target namespace must have the same shard
+	// count as the source namespace in the source cluster. If you do not specify a target namespace, objects are
+	// replicated to a namespace with the same name as the source namespace. If no such namespace exists in the target
+	// cluster, it is created with the same name and shard count as the source namespace. You can only replicate tables
+	// in the public schema to the default_namespace in the target cluster.
+	Namespace string `json:"namespace"`
```

Comment:
```suggestion
	Namespace string `json:"namespace,omitempty"`
```
For backward compatibility.